File: stack/1441_build_an_array_with_stack_operations/1441_build_an_array_with_stack_operations.py
from typing import List


class Solution:
    def buildArray(self, target: List[int], n: int) -> List[str]:
        res = []
        keep = 0
        for num in range(1, n+1):
            res.append("Push")
            keep += 1
            if num not in target:
                res.append("Pop")
                keep -= 1
            if keep == len(target):
                break
        return res


target = [1,3]
n = 3
res = Solution().buildArray(target, n)
print(res)

File: stack/1441_build_an_array_with_stack_operations/1441_build_an_array_with_stack_operations.go
package main

import "fmt"

func buildArray(target []int, n int) []string {
	var (
		ret         []string
		pushedIndex int = 0
	)

	for i := 0; i < len(target); i++ {
		if target[i] == pushedIndex+1 {
			ret = append(ret, "Push")
			pushedIndex++
		}

		if target[i] > pushedIndex+1 {
			for j := 0; j < target[i]-pushedIndex-1; j++ {
				ret = append(ret, "Push")
				ret = append(ret, "Pop")
			}

			pushedIndex += target[i] - pushedIndex - 1
			ret = append(ret, "Push")
			pushedIndex++
		}
	}
	return ret
}

func main() {
	target := []int{1, 3}
	n := 3
	res := buildArray(target, n)
	fmt.Println(res)
}
